ilastik/applets/edgeTraining/opEdgeTraining.py

```diff
@@ -387,37 +387,53 @@ class OpPredictEdgeProbabilities(Operator):
     TrainRandomForest = InputSlot(value=False)
     EdgeClassifier = InputSlot()
     EdgeFeaturesDataFrame = InputSlot()
+    # historically named Edge"Probabilities" because multicut used predictions from
+    # random forest as edge weights. With neural networks, edge probabilities are often already good
+    # enough to act as weights.
     EdgeProbabilities = OutputSlot()  # A 1D array of probabilities, in same order as EdgeFeaturesDataFrame
 
     def setupOutputs(self):
         self.EdgeProbabilities.meta.shape = (1,)
         self.EdgeProbabilities.meta.dtype = object
 
-    def execute(self, slot, subindex, roi, result):
-        pd.set_option("display.expand_frame_repr", False)
-        if self.TrainRandomForest.value:
-            edge_features_df = self.EdgeFeaturesDataFrame.value
-            classifier = self.EdgeClassifier.value
+    def _edge_weights_from_random_forest_predictions(self):
+        edge_features_df = self.EdgeFeaturesDataFrame.value
+        classifier = self.EdgeClassifier.value
 
-            # Classifier can be None if no labels have been selected
-            if classifier is None or len(classifier.known_classes) < 2:
-                result[0] = np.zeros((len(edge_features_df),), dtype=np.float32)
-                return
+        # Classifier can be None if no labels have been selected
+        if len(getattr(classifier, "known_classes", [])) == 0:
+            return np.zeros(len(edge_features_df), dtype=np.float32)
+
+        known_classes = classifier.known_classes
+        assert set(known_classes).issubset([1, 2])
```

Comment:
What are the possible values of `known_classes`? Is it one of `{1}`, `{2}`, and `{1, 2}`?

If yes, I think it'd be better to:

```python
known_classes = set(classifier.known_classes)
if known_classes == {1}:
    return np.zeros(...)
if known_classes == {2}:
    return np.ones(...)
if known_classes == {1, 2}:
    return ...
raise AssertionError(...)
```